pinot-spi/src/main/java/org/apache/pinot/spi/tasks/MinionTaskBaseObserverStats.java

```diff
@@ -103,25 +140,70 @@ public boolean equals(Object o) {
       return false;
     }
     MinionTaskBaseObserverStats stats = (MinionTaskBaseObserverStats) o;
-    return _startTimestamp == stats.getStartTimestamp() && _taskId.equals(stats.getTaskId())
-        && _currentState.equals(stats.getCurrentState());
+    return _startTimestamp == stats.getStartTimestamp() && _endTimestamp == stats.getEndTimestamp()
+        && _taskId.equals(stats.getTaskId()) && _currentState.equals(stats.getCurrentState())
+        && _currentStage.equals(stats.getCurrentStage());
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(_taskId, _currentState, _startTimestamp);
+    return Objects.hash(_taskId, _currentStage, _currentState, _startTimestamp, _endTimestamp);
+  }
+
+  public static class Timer {
+    private long _totalTimeMs = 0;
+    private long _startTimeMs = 0;
+    private long _resumeTimeMs = 0;
+
+    public Timer() {
+    }
+
+    public Timer(@JsonProperty("totalTimeMs") long totalTimeMs,
+        @JsonProperty("startTimeMs") long startTimeMs,
+        @JsonProperty("resumeTimeMs") long resumeTimeMs) {
+      _totalTimeMs = totalTimeMs;
+      _startTimeMs = startTimeMs;
+      _resumeTimeMs = resumeTimeMs;
+    }
+
+    public void start() {
+      _startTimeMs = System.currentTimeMillis();
+      _resumeTimeMs = _startTimeMs;
```

Comment:
What's the real use of the field `_resumeTimeMs`? Do we expect the same stage to started more than once?

Reply:
Yes that's what I was thinking when introducing it. I wanted to keep this flexibility in case we adopt this where a set of stages is ran in a loop, in which case the cumulative stage time will be tracked.